src/dos/drive_local.cpp

```diff
@@ -443,6 +480,18 @@ bool localDrive::GetFileAttr(char *name, uint16_t *attr)
 			*attr |= DOS_ATTR_DIRECTORY;
 		if (!(status.st_mode & S_IWUSR))
 			*attr |= DOS_ATTR_READ_ONLY;
+
+		char xattr[XATTR_FATTR_LEN] = "";
+#ifdef MACOSX
+		int read_len = getxattr(newname, XATTR_FATTR_NAME, xattr, XATTR_FATTR_LEN, 0, 0);
+#else
+		int read_len = getxattr(newname, XATTR_FATTR_NAME, xattr, XATTR_FATTR_LEN);
+#endif
+		if (read_len > 0) {
+			xattr[read_len - 1] = '\0';
```

Comment:
This is off by one, bug hidden because of the quotes.  Should be `xattr[read_len] = '\0` or you over-write the last character.